services/ui_backend_service/data/cache/get_log_file_action.py

```diff
@@ -178,7 +199,87 @@ def get_log_content(task: Task, logtype: str):
         ]
 
 
-def paginated_result(content: List[Tuple[Optional[int], str]], page: int = 1, limit: int = 0, reverse_order: bool = False, output_raw=False):
+class LogProviderBase:
+
+    @staticmethod
+    def get_log_hash(task: Task, logtype: str) -> int:
+        raise NotImplementedError
+
+    @staticmethod
+    def get_log_content(task: Task, logtype: str):
+        raise NotImplementedError
```

Comment:
You could make it `typing.Protocol` which would be arguably a little more elegant, and have get_log_provider return type be protocol type but up to you, don't feel strongly about this.

Reply:
OK, I will add the tests and then try it out.